sklearn/ensemble/tests/test_weight_boosting.py

```diff
@@ -669,3 +673,29 @@ def test_deprecated_base_estimator_parameters_can_be_set():
 
     with pytest.warns(FutureWarning, match="Parameter 'base_estimator' of"):
         clf.set_params(base_estimator__max_depth=2)
+
+
+@pytest.mark.parametrize("algorithm", ["SAMME", "SAMME.R"])
+def test_adaboost_decision_function(algorithm, global_random_seed):
+    """Check that the decision function respect the symmetric constraint for weak
+    learners.
+
+    To check the following, we can train a single weak learner and check that
+    the sum of the decision function is zero.
+
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/26520
+    """
+    X, y = datasets.make_classification(
+        n_classes=3, n_clusters_per_class=1, random_state=global_random_seed
+    )
+    clf = AdaBoostClassifier(
+        n_estimators=1, random_state=global_random_seed, algorithm=algorithm
```

Comment:
By reading the paper snippet quoted in the description of this PR, I would have assumed that this should also hold for `n_estimators > 1`.

Reply:
Indeed.